ssl_permissions.go

```diff
@@ -14,8 +29,58 @@ func sslKeyPermissions(sslkey string) error {
 	if err != nil {
 		return err
 	}
-	if info.Mode().Perm()&0077 != 0 {
-		return ErrSSLKeyHasWorldPermissions
+
+	err = hasCorrectPermissions(info)
+
+	// simplify the error for output purposes
+	if err == ErrSSLKeyHasUnacceptableUserPermissions || err == ErrSSLKeyHasUnacceptableRootPermissions {
```

Comment:
it seems  like it would be simpler if we got rid of `ErrSSLKeyHasUnacceptableUserPermissions` and `ErrSSLKeyHasUnacceptableRootPermissions`. it looks like they are only used here, but always immediately re-written, so no caller would ever see those errors.

alternatively, if there is a reason to keep them, could you leave  a comment explaining why? and also, i think they should be lower-cased so they are not exported.

---

yet another alternative is error-wrapping (see suggestion below)

Reply:
They're used here and in the tests, and ideally would be returned by the function. However, I didn't want to make this change into one that breaks backwards compatibility with existing consumers of the library, which I'd assume would be necessary if I wrapped the error or updating it to a custom error (which would be my preferred alternative to wrapping the errors).

I see your point about not exporting them though! I'll do that now

Reply:
> which I'd assume would be necessary if I wrapped the error

actually wrapping the error should be fine IMO. as i mentioned in https://github.com/lib/pq/pull/1081#discussion_r869882105 calls to `errors.Is(err, ErrSSLKeyHasWorldPermissions)` would still work. 

but it would break any comparisons that use `err == ErrSSLKeyHasWorldPermissions` (which people shouldn't be doing...)

hmmm, so if you feel like it's best as-is, i'm fine to merge

Reply:
Honestly most of the Go code I've worked with still uses `err ==` instead of `errors.Is` so I'd prefer to leave it as is if that's OK with you! I think you might need to run the workflow though, since I'm a first time contributor.

Reply:
ok, then lgtm!